composer/trainer/trainer.py

```diff
@@ -2207,8 +2207,10 @@ def _train_microbatches(self,
                     except TypeError:
                         optimizer.zero_grad()
 
-            # tracker for gradient accumulation
+            # Tracker for gradient accumulation
             current_batch_size = sum([self._train_data_spec.get_num_samples_in_batch(batch) for batch in microbatches])
+            # Cache batch, which will be overwritten by microbatches. Restore after microbatches complete
```

Comment:
Yea, unfortunately we have to cache here as well because we want to restore before `AFTER_TRAIN_BATCH` event. There is another cache elsewhere in `train_microbatch` which is unnecessary though... removing. I think this was a confusion from a merge conflict 6 mo ago...